src/CodeGenerator/src/Generator/ObjectGenerator.php

```diff
@@ -254,13 +253,21 @@ private function addProperties(StructureShape $shape, ClassType $class, PhpNames
                 $deprecation = strtr('@trigger_error(\sprintf(\'The property "NAME" of "%s" is deprecated by AWS.\', __CLASS__), E_USER_DEPRECATED);', ['NAME' => $member->getName()]);
             }
 
-            $method->setBody($deprecation . strtr('
-                    return $this->NAME;
-                ', [
-                'NAME' => $member->getName(),
-            ]));
+            if ($getterSetterNullable) {
+                $method->setBody($deprecation . strtr('
+                        return $this->NAME;
+                    ', [
+                    'NAME' => $member->getName(),
+                ]));
+            } else {
+                $method->setBody($deprecation . strtr('
+                        return $this->NAME ?? [];
+                    ', [
+                    'NAME' => $member->getName(),
+                ]));
+            }
 
-            $nullable = $nullable ?? !$member->isRequired();
+            $nullable = ($nullable ?? !$member->isRequired()) && $getterSetterNullable;
```

Comment:
we still have one nullable use : https://github.com/async-aws/aws/blob/master/src/CodeGenerator/src/Generator/ObjectGenerator.php#L245

Reply:
in fact we still need to distinguish List/Map nullable (to use Null Coalesce in getter) from current nullable (to indicate null in return phpdoc).

Or I miss something ? (monday morning so it could be possible :D)

Reply:
To make is easier to read I could set nullable and getterSetterNullable in List/Map block

Reply:
I updated with my latest proposal